pkg/operator/controller/ingress_status.go

```diff
@@ -0,0 +1,104 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// syncIngressControllerStatus computes the current status of ic and therefrom
+// updates the syncIngressControllerStatus resource.
+func (r *reconciler) syncIngressControllerStatus(deployment *appsv1.Deployment, ic *operatorv1.IngressController) error {
+	selector, err := metav1.LabelSelectorAsSelector(deployment.Spec.Selector)
+	if err != nil {
+		return fmt.Errorf("ingress controller deployment %s/%s has invalid spec.selector: %v", deployment.Namespace, deployment.Name, err)
+	}
+
+	oldStatus := ic.Status.DeepCopy()
+	oldStatus.AvailableReplicas = deployment.Status.AvailableReplicas
+	oldStatus.Selector = selector.String()
```

Comment:
I think you want to assign `ic.Status.AvailableReplicas` and `ic.Status.Selector` here (and then you can delete the assignments to those fields inside the `if` below).

Reply:
I have updated the logic. I now create a `DeepCopy`, assign the status fields, compare the copy and the original ingress for equality. If the two are not equal, I update the status through the api using the copy.